Duplicati/Library/Common/IO/SystemIOWindows.cs

```diff
@@ -29,20 +29,47 @@ namespace Duplicati.Library.Common.IO
 {
     public struct SystemIOWindows : ISystemIO
     {
-        private const string UNCPREFIX = @"\\?\";
-        private const string UNCPREFIX_SERVER = @"\\?\UNC\";
-        private const string PATHPREFIX_SERVER = @"\\";
+        private const string UNCPREFIX = PathInternalWindows.ExtendedDevicePathPrefix;
+        private const string UNCPREFIX_SERVER = PathInternalWindows.UncExtendedPathPrefix;
+        private const string PATHPREFIX_SERVER = PathInternalWindows.UncPathPrefix;
+        private const string PATHPREFIX_SERVER_ALT = @"//";
         private static readonly string DIRSEP = Util.DirectorySeparatorString;
 
+        /// <summary>
+        /// Prefix path with one of the UNC prefixes, but only if it's a fully
+        /// qualified path with no relative components (i.e., with no "." or
+        /// ".." as part of the path).
+        /// </summary>
         public static string PrefixWithUNC(string path)
         {
             if (IsPrefixedWithUNC(path))
             {
+                // For example: \\?\C:\Temp\foo.txt or \\?\UNC\example.com\share\foo.txt
+                return path;
+            }
+            else if (IsPrefixedWithBasicUNC(path) && !HasRelativePathComponents(path))
+            {
+                // For example: \\example.com\share\foo.txt or //example.com/share/foo.txt
+                return UNCPREFIX_SERVER + ConvertSlashes(path.Substring(PATHPREFIX_SERVER.Length));
+            }
+            else if (DotNetRuntimePathWindows.IsPathFullyQualified(path) && !HasRelativePathComponents(path))

```

Comment:
Minor optimization, but we can assign `HasRelativePathComponents(path)` to a variable to avoid possible multiple enumeration of the path components.

Reply:
Yes.  I've pushed a commit with this change.